### PR TITLE
NDK event migration v4 tests and fixes

### PR DESCRIPTION
**Pull request by kattrali**
## Goal

_Don't panic, this changeset is 90% tests_

1. Fix fix omissions (context) and mis-sized memcpys (breadcrumbs, etc) in v4 migration 
2. Add new testing strategy that should be simpler to use going forward
3. Fix feature flags being an object (stolen from #1550, but with impact on this test)

## Design

Introduced a new testing style so that assertions are more visible to the developer while running the tests, and so the event generator code and expected event output can be compared side-by-side easily. Left the existing tests as-is since they are more unit-y and these new tests verify event writing and json serialization together.

## Testing

Added new tests for migration v4, with more to follow for later migrations in subsequent changeset.

### CODE REVIEW

File: bugsnag-plugin-android-ndk/detekt-baseline.xml
Comment (kattrali) [inline bugsnag-plugin-android-ndk/detekt-baseline.xml:6]: This seemed like a sensible long method, since it verifies several things and pulling the objects out of the test scope didn't make sense. Though there will be another "issue" for every test in this format. 🤷

File: bugsnag-plugin-android-ndk/src/androidTest/java/com/bugsnag/android/ndk/migrations/EventMigrationV4Tests.kt
Comment (kattrali) [inline bugsnag-plugin-android-ndk/src/androidTest/java/com/bugsnag/android/ndk/migrations/EventMigrationV4Tests.kt:21]: This info is persisted but currently wholly unused - this key _should_ be used to send the event, but is not afaict.

File: bugsnag-plugin-android-ndk/src/androidTest/java/com/bugsnag/android/ndk/migrations/EventMigrationTest.kt
Comment (lemnik) [inline bugsnag-plugin-android-ndk/src/androidTest/java/com/bugsnag/android/ndk/migrations/EventMigrationTest.kt:12]: This doesn't appear to be re-assigned anywhere, so maybe it can be a `val`?
```suggestion
    private val objectMapper = ObjectMapper()
```

File: bugsnag-plugin-android-ndk/build.gradle
Comment (fractalwrench) [inline bugsnag-plugin-android-ndk/build.gradle:15]: It feels preferable to use dsl-json or GSON rather than bringing another JSON serializer into the project, even if it's just for test code

**Review by fractalwrench — APPROVED** [review]
LGTM pending one review comment

File: bugsnag-plugin-android-ndk/build.gradle
Comment (fractalwrench) [reply bugsnag-plugin-android-ndk/build.gradle:15]: e.g. 

```
byte[] input;
DslJson<Object> dslJson = new DslJson<Object>();
Map<String, Object> map = dslJson.deserialize(Map.class, input, input.length);
```

**Review by lemnik — APPROVED** [review]
LGTM

File: bugsnag-plugin-android-ndk/build.gradle
Comment (kattrali) [reply bugsnag-plugin-android-ndk/build.gradle:15]: I attempted to pull in GSON (since we only have the streaming bits vendored in), but its giving me some JSON validity warnings for reasons that don't seem worth chasing for tests. dsl-json does not (yet?) exist in the repo this juncture, but I can fix it after the next merge or somthing to use `JsonHelper`.

Comment (fractalwrench) [reply bugsnag-plugin-android-ndk/build.gradle:15]: Yeah - it'd be worth merging in `next` if the integration branch is behind, as that'd allow this to use `JsonHelper`.

Comment (kattrali) [reply bugsnag-plugin-android-ndk/build.gradle:16]: Its a non-trivial merge (with a few conflicts in `Event` classes), so it might be worth holding until I get the tests in. I've got tests for all of the subsequent migrations and JSON serialization ready to go.

**Pull request merged**
